compiler/luci/pass/src/FuseMulWithFullyConnectedPass.test.cpp

```diff
@@ -0,0 +1,248 @@
+ *
+ *  BEFORE
+ *
+ *         [FC]
+ *           |
+ *     [Mul w/ Relu]
+ *
+ *  AFTER
+ *
+ *      [FC w/ Relu] (weights and bias updated)
+ *
+ */
+class FCMulGraphlet
+{
+public:
+  void init(loco::Graph *g, luci::FusedActFunc fc_activation, bool is_mul_scalar, bool use_bias)
+  {
+    _fc = g->nodes()->create<luci::CircleFullyConnected>();
+
+    std::vector<float> weights_val(DIM_ONE * DIM_TWO);
+    for (uint32_t i = 0; i < DIM_ONE * DIM_TWO; i++)
+      weights_val.at(i) = i;
+
+    _fc_f = luci::create_const_node(g, loco::DataType::FLOAT32, {DIM_ONE, DIM_TWO}, weights_val);
+    _fc->weights(_fc_f);
+
+    if (use_bias)
+    {
+      std::vector<float> bias_val(DIM_ONE);
+      for (uint32_t i = 0; i < DIM_ONE; i++)
+        bias_val.at(i) = i;
+
+      _fc_b = luci::create_const_node(g, loco::DataType::FLOAT32, {DIM_ONE}, bias_val);
+      _fc->bias(_fc_b);
+    }
+
+    _fc->fusedActivationFunction(fc_activation);
+    _fc->dtype(loco::DataType::FLOAT32);
+    _fc->shape({1, DIM_ONE});
+    _fc->name("fc");
+
+    std::vector<float> mul_values;
+
+    if (is_mul_scalar)
+    {
+      mul_values.push_back(static_cast<float>(MUL_VAL));
+      _mul_c = luci::create_const_node(g, loco::DataType::FLOAT32, {}, mul_values);
+    }
+    else
+    {
+      for (uint32_t i = 0; i < DIM_ONE; i++)
+      {
+        mul_values.push_back(static_cast<float>(i));
+      }
+      _mul_c = luci::create_const_node(g, loco::DataType::FLOAT32, {1, 1, 1, DIM_ONE}, mul_values);
+    }
+
+    _mul = g->nodes()->create<luci::CircleMul>();
+    _mul->x(_fc);
+    _mul->y(_mul_c);
+    _mul->fusedActivationFunction(luci::FusedActFunc::RELU);
+    _mul->dtype(loco::DataType::FLOAT32);
+    if (is_mul_scalar)
+    {
+      _mul->shape({1});
+    }
+    else
+    {
+      _mul->shape({1, DIM_ONE});
+    }
+    _mul->name("mul");
+  }
+
+public:
+  luci::CircleFullyConnected *fc() { return _fc; }
+
+  void to_fm_bias(void)
+  {
+    assert(_fc != nullptr);
+
+    auto new_fc = _fc->graph()->nodes()->create<luci::CircleFullyConnected>();
+    _fc->bias(new_fc);
+  }
+
+protected:
+  luci::CircleFullyConnected *_fc = nullptr;
+  luci::CircleMul *_mul = nullptr;
+  luci::CircleConst *_fc_f = nullptr;
+  luci::CircleConst *_fc_b = nullptr;
+  luci::CircleConst *_mul_c = nullptr;
+};
+
+class FuseMulWithFCTestGraph : public TestIOGraph, public FCMulGraphlet
+{
+public:
+  void init(luci::FusedActFunc fc_activation, bool is_mul_scalar, bool use_bias)
+  {
+    TestIOGraph::init({1, DIM_TWO}, {1, DIM_ONE});
+    FCMulGraphlet::init(g(), fc_activation, is_mul_scalar, use_bias);
+
+    _fc->input(input());
+
+    output()->from(_mul);
+  }
+};
+
+class FuseMulWithFullyConnectedPassTest : public ::testing::Test
+{
+public:
+  FuseMulWithFCTestGraph g;
+  luci::FuseMulWithFullyConnectedPass pass;
+};
+
+} // namespace
+
+TEST_F(FuseMulWithFullyConnectedPassTest, fc_without_activation_mul_not_scalar)
+{
+  g.init(luci::FusedActFunc::NONE, false, true);
```

Comment:
Could you add comments for readability?

```suggestion
  g.init(luci::FusedActFunc::NONE, false /* is_mul_scalar */, true /* use_bias */);
```

Reply:
Done for all tests.